src/features/MonthSlots/helpers.ts

```diff
@@ -1,37 +1,75 @@
-// export const generateSlotsForDaysOfMonth = (
-//   currentYear: any,
-//   daysInMonth: any,
-//   currentMonth: any,
-//   slotsData: any,
-// ): {
-//   slots: any
-//   date: Date
-// }[] => {
-//   const cells = Array.from({ length: daysInMonth }, (_, day) => ({
-//     date: new Date(currentYear, currentMonth, day + 1),
-//     slots: [],
-//   }))
-
-//   const newCells = cells.map(cell => {
-//     const matchingSlotData = slotsData.find(({ date }: any) => {
-//       const slotDate = new Date(date)
-
-//       return (
-//         slotDate.getMonth() === currentMonth &&
-//         slotDate.getFullYear() === currentYear &&
-//         slotDate.getDate() === cell.date.getDate()
-//       )
-//     })
-
-//     if (matchingSlotData) {
-//       return {
-//         ...cell,
-//         slots: matchingSlotData.slots,
-//       }
-//     }
-
-//     return cell
-//   })
-
-//   return newCells
-// }
+import { CalendarEventType } from '../../types'
+
+import { Cell } from './types'
+import { countCells } from './constants'
+
+export const createCells = (
+  currentYear: number,
+  currentMonth: number,
+  countCells: number,
+  isCurrentMonth: boolean,
+  daysInPrevMonth?: number,
+): Cell[] =>
+  Array.from({ length: countCells }, (_, day) => {
+    return {
+      date: new Date(
+        currentYear,
+        currentMonth,
+        daysInPrevMonth ? daysInPrevMonth - day : day + 1,
+      ),
+      isCurrentMonth,
+      slots: [],
+    }
+  })
+
+export const generateSlotsForDaysOfMonth = (
+  currentYear: number,
+  daysInMonth: number,
+  currentMonth: number,
+  slotsData: CalendarEventType[],
+  firstDayOfMonth: number,
+): Cell[] => {
```

Comment:
I would personally pass those arguments as an object. So you don't have to keep in mind the right sequence of the arguments